WordPress/gutenberg
# Use a specific CSS class for the CSS reset mixin.

## Description
After #14509, the CSS rules for form controls target also the metaboxes area, conflicting with styles provided by plugins.

See  for example https://github.com/Yoast/wordpress-seo/issues/13316

For now, the new form controls styles are used only in Gutenberg. Eventually, when these styles will be used in core, WordPress would expect plugins to honor the new styles but their CSS selectors _would need to have a lower specificity_.

In fact, WordPress styles form controls using CSS selectors with a relatively low specificity. For example: `input[type="text"]`. Instead, the selectors used in Gutenberg need to have a higher specificity. By doing so, their associated CSS rules tend to override the CSS provided by plugins.

Once the new form controls styles will be in core, this problem will go away. For now, I'd like to propose to reinforce the principle that Gutenberg should do its best to not create conflicts with CSS provided by plugins. 

This PR seeks to better scope the `reset` mixin:
- introduces a new CSS class `block-editor-styles-reset`: its sole responsibility is to apply the `reset` mixin
- sets the `block-editor-styles-reset` class on the relevant places in the UI

Open to better naming for the new CSS class, of course.

**Added bonus**
_If_ plugins want to use the new form controls styles, they can opt for that by simply applying the new CSS class on their UI.


**Test instructions:**
Check all form controls have the new styles in the following places in the UI:
- the post content with the list of blocks (e.g. insert a Table block and check the input fields for the number of columns/rows)
- the post permalink UI
- the normal sidebar
- the publish sidebar
- all the popovers (e.g. the inserter or the publish date one)
- all the modal dialogs (e.g. the Block Manager or the Options one)
- install / activate one or more plugins that render some UI in the metaboxes area (e.g. Advanced Custom Fields, Yoast SEO)
- verify the new Gutenberg form controls styles do not apply to the plugins UI within the metaboxes  area
- activate the "Custom Fields" panel from the "Options" Gutenberg modal dialog
- verify the new Gutenberg form controls styles do not apply to the Custom Fields area
